pkg/vault/contracts/BatchRouter.sol

```diff
@@ -671,6 +671,317 @@ contract BatchRouter is IBatchRouter, BatchRouterStorage, RouterCommon, Reentran
+        (amountsIn, , ) = abi.decode(
+            _vault.quote(
+                abi.encodeWithSelector(
+                    BatchRouter.addLiquidityERC4626PoolProportionalHook.selector,
+                    AddLiquidityHookParams({
+                        sender: msg.sender,
+                        pool: pool,
+                        maxAmountsIn: maxUnderlyingAmountsIn,
+                        minBptAmountOut: exactBptAmountOut,
+                        kind: AddLiquidityKind.PROPORTIONAL,
+                        wethIsEth: false,
+                        userData: userData
+                    })
+                )
+            ),
+            (uint256[], uint256, bytes)
+        );
+    }
+
+    /// @inheritdoc IBatchRouter
+    function queryRemoveLiquidityProportionalToERC4626Pool(
+        address pool,
+        uint256 exactBptAmountIn,
+        bytes memory userData
+    ) external saveSender returns (uint256[] memory amountsOut) {
+        (, amountsOut, ) = abi.decode(
+            _vault.quote(
+                abi.encodeWithSelector(
+                    BatchRouter.removeLiquidityERC4626PoolProportionalHook.selector,
+                    RemoveLiquidityHookParams({
+                        sender: msg.sender,
+                        pool: pool,
+                        minAmountsOut: new uint256[](2),
+                        maxBptAmountIn: exactBptAmountIn,
+                        kind: RemoveLiquidityKind.PROPORTIONAL,
+                        wethIsEth: false,
+                        userData: userData
+                    })
+                )
+            ),
+            (uint256, uint256[], bytes)
+        );
+    }
+
+    function addLiquidityERC4626PoolUnbalancedHook(
+        AddLiquidityHookParams calldata params
+    )
+        external
+        nonReentrant
+        onlyVault
+        returns (uint256[] memory amountsIn, uint256 bptAmountOut, bytes memory returnData)
+    {
+        IERC20[] memory erc4626PoolTokens = _vault.getPoolTokens(params.pool);
+        uint256[] memory wrappedAmounts = _wrapToken(
+            params,
+            erc4626PoolTokens,
+            params.maxAmountsIn,
+            SwapKind.EXACT_IN,
+            0
+        );
+
+        // Add wrapped amounts to the boosted pool
+        (amountsIn, bptAmountOut, returnData) = _vault.addLiquidity(
+            AddLiquidityParams({
+                pool: params.pool,
+                to: params.sender,
+                maxAmountsIn: wrappedAmounts,
+                minBptAmountOut: params.minBptAmountOut,
+                kind: params.kind,
+                userData: params.userData
+            })
+        );
+    }
+
+    function addLiquidityERC4626PoolProportionalHook(
+        AddLiquidityHookParams calldata params
+    )
+        external
+        nonReentrant
+        onlyVault
+        returns (uint256[] memory amountsIn, uint256 bptAmountOut, bytes memory returnData)
```

Comment:
Same; since this is specific to proportional add, maybe returning `amountsIn` is enough.